packages/core/src/system.ts

```diff
@@ -173,11 +207,56 @@ export function createSystem<T extends ActorSystemInfo>(
       if (systemId !== undefined) {
         keyedActors.delete(systemId);
         reverseKeyedActors.delete(actorRef);
+        const event = {
+          type: `@xstate.system.actor.unregister.${systemId as string}`,
+          actorRef: actorRef as T['actors'][keyof T['actors']]
+        } as const;
+        systemSubscriptions.get(systemId)?.forEach((listener) => {
+          listener(event);
+        });
+        systemSubscriptions.get(rootSubscriptionKey)?.forEach((listener) => {
+          listener(event);
+        });
       }
     },
     get: (systemId) => {
       return keyedActors.get(systemId) as T['actors'][any];
     },
+    subscribe: <K extends keyof T['actors']>(
+      maybeSystemIdOrListener:
+        | K
+        | ((event: RegistrationEvent<T['actors'][keyof T['actors']]>) => void),
+      maybeListener?:
+        | ((event: RegistrationEvent<T['actors'][keyof T['actors']]>) => void)
+        | undefined
+    ) => {
+      let listener =
+        typeof maybeSystemIdOrListener === 'function'
+          ? maybeSystemIdOrListener
+          : maybeListener!;
+      let systemId =
+        typeof maybeSystemIdOrListener === 'function'
+          ? rootSubscriptionKey
+          : maybeSystemIdOrListener;
+
+      let subscriptions = systemSubscriptions.get(systemId) || new Set();
```

Comment:
I think we could also design this differently by using a single set, and overriding the listeners if a systemId key is present. This would also mean passing the systemId directly in the event so its easier to parse

something like:

```tsx
let listener = 
  typeof maybeSystemIdOrListener === 'function' ? 
  maybeSystemIdOrListener 
  : 
    (event) => { 
      if (event.systemId === maybeSystemIdOrListener) maybeListener(event)
     }
```

what do you think is preferable here?